collect_app/src/main/java/org/odk/collect/android/widgets/utilities/RangeWidgetDataRequester.java

```diff
@@ -0,0 +1,186 @@
+            currentValue = rangeWidgetVerticalBinding.currentValue;
+            minValue = rangeWidgetVerticalBinding.minValue;
+            maxValue = rangeWidgetVerticalBinding.maxValue;
+        } else {
+            RangeWidgetHorizontalBinding rangeWidgetHorizontalBinding = RangeWidgetHorizontalBinding
+                    .inflate(((Activity) context).getLayoutInflater());
+            answerView = rangeWidgetHorizontalBinding.getRoot();
+
+            slider = rangeWidgetHorizontalBinding.slider;
+            currentValue = rangeWidgetHorizontalBinding.currentValue;
+            minValue = rangeWidgetHorizontalBinding.minValue;
+            maxValue = rangeWidgetHorizontalBinding.maxValue;
+        }
+        return new RangeWidgetLayoutElements(answerView, slider, currentValue, minValue, maxValue);
+    }
+
+    public static BigDecimal getActualValue(FormEntryPrompt prompt, Slider slider, BigDecimal rangeStart,
+                                            BigDecimal rangeEnd, BigDecimal rangeStep, BigDecimal actualValue) {
+
+        BigDecimal startValue =  rangeEnd.compareTo(rangeStart) < 0 ? rangeEnd : rangeStart;
+
+        if (WidgetAppearanceUtils.hasAppearance(prompt, NO_TICKS_APPEARANCE)) {
+            int progress = (actualValue.subtract(startValue).abs().divide(rangeStep)).intValue();
+            actualValue = startValue.add(startValue.multiply(BigDecimal.valueOf(progress)));
+
+            slider.setValue(actualValue.floatValue());
+        }
+
+        if (rangeEnd.compareTo(rangeStart) < 0) {
+            actualValue = rangeEnd.add(rangeStart).subtract(actualValue);
+        }
+        return actualValue;
+    }
+
+    public static String[] getDisplayedValuesForNumberPicker(BigDecimal rangeStart, BigDecimal rangeStep,
+                                                             BigDecimal rangeEnd, Boolean isIntegerDataType) {
+        int index = 0;
+        int elementCount = rangeEnd.subtract(rangeStart).abs().divide(rangeStep).intValue();
+        String[] displayedValuesForNumberPicker = new String[elementCount + 1];
+
+        if (rangeEnd.compareTo(rangeStart) > -1) {
+            for (BigDecimal i = rangeEnd; i.compareTo(rangeStart) > -1; i = i.subtract(rangeStep.abs())) {
+                displayedValuesForNumberPicker[index] = isIntegerDataType ? String.valueOf(i.intValue())
+                        : String.valueOf(i.doubleValue());
+                index++;
+            }
+        } else {
+            for (BigDecimal i = rangeEnd; i.compareTo(rangeStart) < 1; i = i.add(rangeStep.abs())) {
+                displayedValuesForNumberPicker[index] = isIntegerDataType ? String.valueOf(i.intValue())
+                        : String.valueOf(i.doubleValue());
+                index++;
+            }
+        }
+        return displayedValuesForNumberPicker;
+    }
+
+    public static void requestRangePickerValue(Context context, WaitingForDataRegistry waitingForDataRegistry, FormIndex formIndex,
```

Comment:
I think this has all been a little mixed up. To me, it feels like the `RangeWidgetDataRequester` should be an interface with the `requestRangePickerValue` method on it and there should be an implementation that uses the ViewModel/DialogFragment. In my head this should work like the `GeoDataRequester`. Does that make sense?

Happy to chat about this more on Slack or somewhere if that's helpful.

Reply:
Based on our discussion in the other comment we  don't want the data requester to use the viewmodel but it should still be reworked as an interface (and have the static methods split out). Also the `WaitingForDataRegistry` and `Content` could be passed in at the constructor level rather than in this method.